contracts/libraries/LockDataLibrary.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+
+library LockDataLibrary {
```

Comment:
imo think this could use a comment explaining the custom storage layout handling

Reply:
semi scary as you get into potential collision territory ie if using this lib on two diff instances of LockData or something dumb haha

Reply:
btw how much is the gas advantage over just having 
```
struct LockData {
  ...
  Locker[] lockers;
}
```

Reply:
yeah i don't love that this lib can be used on two different LockDatas either but i couldn't really think of a clean way to limit this behavior. one idea i had is to write a "custom storage offsets" lib to consolidate all of the offsets and then use values from this lib either directly in consumer libs or pass them through directly as an argument but this also felt needlessly verbose. i think i'm in favor of just being careful 😬

as far as gas, we save a fair amount, as we're able to pack the active locker and the non-0 delta count with the length of the array, so there are a lot of saved sstores/sloads. happy to do comps but it's definitely going to be a fair amount

Reply:
Core doesn't have two different LockDatas though right? So not an issue persay but just not desireable?